store/store.go

```diff
@@ -28,12 +29,61 @@ type Store struct {
 	closed     bool
 }
 
-type rpcSubscriptionWrapper struct {
-	*rpc.Client
+type lazyRPCWrapper struct {
+	client *rpc.Client
+	url    *url.URL
+	mutex  *sync.Mutex
 }
 
-func (wrapper rpcSubscriptionWrapper) EthSubscribe(ctx context.Context, channel interface{}, args ...interface{}) (models.EthSubscription, error) {
-	return wrapper.Client.EthSubscribe(ctx, channel, args...)
+func newLazyRPCWrapper(urlString string) (CallerSubscriber, error) {
+	parsed, err := url.ParseRequestURI(urlString)
+	if err != nil {
+		return nil, err
+	}
+	if parsed.Scheme != "ws" && parsed.Scheme != "wss" {
+		return nil, fmt.Errorf("Ethereum url scheme must be websocket: %s", parsed.String())
+	}
+	return &lazyRPCWrapper{
+		url:   parsed,
+		mutex: &sync.Mutex{},
+	}, nil
+}
+
+// lazyDialInitializer initializes the Dial instance used to interact with
+// an ethereum node using the Double-checked locking optimization:
+// https://en.wikipedia.org/wiki/Double-checked_locking
+func (wrapper *lazyRPCWrapper) lazyDialInitializer() error {
+	if wrapper.client != nil {
```

Comment:
Does the race detector complain about this?

Reply:
It didn't, but I think it's a data race issue nonetheless. I've committed a fix.